File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.c
int maxNumberOfBalloons(char * text){
    int arr[5] = {0};  //b,a,l,o,n
    int res = 0;
    
    for(int i = 0; text[i]; i++)
    {
        if     (text[i] == 'b') arr[0]++;
        else if(text[i] == 'a') arr[1]++;
        else if(text[i] == 'l') arr[2]++;
        else if(text[i] == 'o') arr[3]++;
        else if(text[i] == 'n') arr[4]++;
    }
    
    while(arr[0] > 0 && arr[1] > 0 &&
          arr[2] > 1 && arr[3] > 1 && arr[4] > 0)
    {
        arr[0]--;
        arr[1]--;
        arr[2] -= 2;
        arr[3] -= 2;
        arr[4]--;
        res++;
    }
    return res;
}

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.swift
class Solution {
    func maxNumberOfBalloons(_ text: String) -> Int {
        var arr = [0,0,0,0,0]   // b,a,l,o,n
        var cnt = 1_000_000_000
        
        for char in text {
            if char == "b" { arr[0] += 1 }
            else if char == "a" { arr[1] += 1 }
            else if char == "l" { arr[2] += 1 }
            else if char == "o" { arr[3] += 1 }
            else if char == "n" { arr[4] += 1 }
        }
        
        var i = 0
        while i < arr.count {
            if arr[i] == 0 { return 0 }
            if i == 2 || i == 3 {
                if arr[i] / 2 < cnt {
                    cnt = arr[i] / 2
                }
            } else {
                if arr[i] < cnt { 
                    cnt = arr[i]
                }
            }
            i += 1
        }
        
        return cnt
    }
}